draft-irtf-cfrg-vdaf.md

```diff
@@ -2826,6 +2870,22 @@ def prep_shares_to_prep(Poplar1, agg_param, prep_shares):
 ~~~
 {: #poplar1-prep-state title="Preparation state for Poplar1."}
 
+### Validity of Aggregation Parameters
+
+Aggregation parameters are valid for a given input share, if no aggregation
+parameter with the same level has been used with the same input share before.
+The whole preparation phase MUST NOT be run more than once for a given
+combination of input share and level.
+
+~~~
+def is_valid(agg_param, previous_agg_params):
+    (level, _) = agg_param
+    return len(filter(lambda (other_level, _): other_level == level, \
+                     previous_agg_params)) == 0
```

Comment:
This may be more idiomatic and clearer.
```suggestion
    return all(
        level != other_level
        for (other_level, _) in previous_agg_params
    )
```

Reply:
Done.